src/node.cc

```diff
@@ -847,6 +861,13 @@ void PromiseRejectCallback(PromiseRejectMessage message) {
 
   if (event == v8::kPromiseRejectWithNoHandler) {
     callback = env->promise_reject_unhandled_function();
+
+    Persistent<Promise>* p = new Persistent<Promise>(env->isolate(), promise);
+
+    p->SetWeak(p, WeakCallback, WeakCallbackType::kFinalizer);
```

Comment:
Shouldn’t there be a `ClearWeak()` call somewhere, in case the rejection is handled?

Reply:
Yes, I forgot about that. Thanks a lot for pointing it out :-)